DOCSP-37722: Quick start guide (#5)
mongodb/docs-java-rs: source/get-started/quickstart.txt

   automatically generates a value and adds this field to the inserted
   document.

.. important::

   In the API, all methods returning a ``Publisher`` instance are cold
   streams, meaning that nothing happens until they are subscribed to.

   For example, the example below does nothing:
   
   .. code-block:: java
   
      Publisher<InsertOneResult> publisher = collection.insertOne(doc);

   The operation happens only when a ``Publisher`` is subscribed to and
   data is requested.

   .. code-block:: java
         
      publisher.subscribe(new Subscriber<InsertOneResult>() {
          @Override
          public void onSubscribe(final Subscription s) {
              s.request(1);  // <--- Data requested and the insertion will now occur
          }
      
          @Override
          public void onNext(final InsertOneResult result) {
              System.out.println("Inserted: " + result);
          }
      
          @Override
          public void onError(final Throwable t) {
              System.out.println("Failed");
          }
      
          @Override
          public void onComplete() {
              System.out.println("Completed");
          }
      });

>    Once the document has been inserted the ``onNext()`` method will be called

Reviewer: ```suggestion
   Once the document has been inserted, the ``onNext()`` method will be called
```

>    and print ``Inserted:`` followed by the result. Finally, the

Reviewer: ```suggestion
   and will print ``Inserted:`` followed by the result. Finally, the
```

   ``onComplete()`` method will print ``Completed``. If there was an error

>    for any reason the ``onError()`` method would print ``Failed``.

Reviewer: ```suggestion
   for any reason, the ``onError()`` method would print ``Failed``.
```

Insert Multiple Documents
~~~~~~~~~~~~~~~~~~~~~~~~~

To insert multiple documents, you can use the collection's
``insertMany()`` method, which takes a list of documents to insert.

The following example adds multiple documents in the following form:

.. code-block:: json

   { "i" : value }

Create the documents in a loop and add them to the ``documents`` list:

.. code-block:: java

   List<Document> documents = new ArrayList<Document>();
   for (int i = 0; i < 100; i++) {
       documents.add(new Document("i", i));
   }

To insert these documents into the collection, pass the list of
documents to the ``insertMany()`` method:

.. code-block:: java

   var subscriber = new OperationSubscriber<InsertManyResult>();
   collection.insertMany(documents).subscribe(subscriber);
   
   // Wait for the insertMany operation to complete
   subscriber.await();

> Here, we block on the ``Publisher`` to finish so that when we call the
> next operation we know the data has been inserted into the database.

Reviewer: ```suggestion
The previous example blocks on the ``Publisher`` to finish. This ensures
that the data is in the database before the next operation runs.
```
